File: class_tool/src/main/java/com/tempsfteam/class_tool/entity/Profession.java
package com.tempsfteam.class_tool.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.tempsfteam.class_tool.dto.ProfessionDTO;
import lombok.Data;

import java.io.Serializable;

/**
 * @author hypocodeemia
 * @TableName profession
 */
@TableName(value ="profession")
@Data
public class Profession implements Serializable {
    @TableId(type = IdType.AUTO)
    private Integer professionId;

    private Integer courseTypeId;

    private String name;

    private Integer sort;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;

    public Profession() {
    }

    public Profession(Integer courseTypeId, String name) {
        this.courseTypeId = courseTypeId;
        this.name = name;
    }

    public Profession(ProfessionDTO professionDTO) {
        this.professionId = professionDTO.getProfessionId();
        this.courseTypeId = professionDTO.getCourseTypeId();
        this.name = professionDTO.getName();
    }
}

File: class_tool/src/main/resources/static/src/models/entity/profession.js
export default {
    name: 'profession',
    title: '专业',
    idField: 'professionId',
    fields: [
        { name: 'professionId', label: '专业ID', type: 'number', readonly: true },
        { name: 'name', label: '专业名', type: 'string', required: true },
        { name: 'courseTypeId', label: '课程类型ID', type: 'number' },
        { name: 'sort', label: '排序', type: 'number' }
    ]
};